### ISSUE TITLE
XMLHttpRequest: "TypeError: Illegal invocation" when setting custom timeout

### ISSUE THREAD

**Issue by romaia** [mswjs/interceptors#339]
I was trying to use this with axios, and didn't get it to work. Upon further inspection, I figured the issue is with `XMLHttpRequest.timeout`.

The following case ilustrates the issue:

```js
const req = new XMLHttpRequest();
req.timeout = 10
req.open("GET", "http://www.example.org/example.txt");
req.send();
```

I get this debug log and error

```
common.js:113 xhr constructed new XMLHttpRequest +8m
createProxy.ts:46 Uncaught TypeError: Illegal invocation
    at Reflect.set (<anonymous>)
    at next (createProxy.ts:46:34)
    at XMLHttpRequest.setProperty (XMLHttpRequestController.ts:61:16)
    at Object.handler.set (createProxy.ts:47:26)
    at <anonymous>:2:13
```
Removing the `req.timeout = 10` line makes que request go all the way.

**Comment by kettanaito** [comment]
Hey, @romaia. Thanks for reporting this! 

Looks like it's the issue in the proxy we create. I initially assumed it to be related to the `withCredentials` property but it seems that it affects any setters on the XMLHttpRequest object. 

We need to look into why setters are causing the illegal invocation error and resolve it.

This likely means rewriting this fix:

https://github.com/mswjs/interceptors/blob/29acc5998911c36059e19bb1d6d8d62d3d64ea6d/src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts#L50-L62

And simply using that `define()` call on all XMLHttpRequest setters. 

Would you be interested in opening a pull request with that change? I can help you write tests for this.

**Comment by romaia** [comment]
Seems like this is actually a regression. Testing with the previous version is working.

**Comment by kettanaito** [comment]
It is a regression as 0.20.0 introduces a new interception algorithm for XMLHttpRequest.

**Comment by kettanaito** [comment]
## Released: v0.22.4 🎉

This has been released in v0.22.4!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.22.4)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.22.4)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
